### PR TITLE
Node 2498/polyfill promisify

### PR DESCRIPTION
**Pull request by reggi**
## Description

**What changed?**

Pollyfill for `util.promisify` due to removal of `bluebird`.

**Are there any files to ignore?**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,10 +28,10 @@
     "bson": "^1.1.1",
     "denque": "^1.4.1",
     "require_optional": "^1.0.1",
-    "safe-buffer": "^5.1.2",
-    "util.promisify": "^1.0.1"
+    "safe-buffer": "^5.1.2"
   },
   "devDependencies": {
+    "util.promisify": "^1.0.1",
```
Comment (mbroadst) [inline package.json]: nit: there's a marginal effort to keep this list alphabetically sorted

**Comment by mbroadst** [comment]
@reggi I can't comment on this, but it looks like `kerberos` is being brought in as a git submodule in this PR. What's up there?

File: package.json (hunk 1 above)
Comment (reggi) [reply package.json]: 👍

File: test/unit/cmap/connection_pool.test.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 'use strict';
 
+require('util.promisify/shim')();
```
Comment (emadum) [inline test/unit/cmap/connection_pool.test.js]: Minor nitpick, but I'd put this next to the `require('util')` just for the sake of organization.

Comment (reggi) [reply test/unit/cmap/connection_pool.test.js]: Hey, this was brought closer to the util call.

Comment (emadum) [reply test/unit/cmap/connection_pool.test.js]: Thanks!

**Pull request merged**